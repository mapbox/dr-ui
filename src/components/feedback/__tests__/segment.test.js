const { createSegmentEvent } = require('../segment.js');

describe('createSegmentEvent', () => {
  describe('anonymous user', () => {
    test('1 - Open feedback', () => {
      expect(
        createSegmentEvent({
          state: {
            anonymousId: '7a6b4852-bacf-40e1-ae42-d91c4003f464',
            sessionId: '1624292376749-7a6b4852-bacf-40e1-ae42-d91c4003f464',
            isOpen: true,
            sentFeedback: false,
            contactSupport: false
          },
          props: {
            site: 'dr-ui',
            webhook: { staging: 'webhook', production: 'webhook' },
            preferredLanguage: 'Swift',
            section: 'LngLat',
            location: { pathname: '/dr-ui/feedback/', hash: '#lnglat' },
            type: 'page',
            feedbackSentryDsn: 'https://abce@sentry.io/123'
          }
        })
      ).toMatchSnapshot();
    });

    test('2 - Report a problem', () => {
      expect(
        createSegmentEvent({
          state: {
            anonymousId: '7a6b4852-bacf-40e1-ae42-d91c4003f464',
            sessionId: '1624292376749-7a6b4852-bacf-40e1-ae42-d91c4003f464',
            isOpen: true,
            category: 'Report a problem',
            sentFeedback: false,
            contactSupport: false,
            helpful: false
          },
          props: {
            site: 'dr-ui',
            webhook: {
              staging: 'webhook',
              production: 'webhook'
            },
            preferredLanguage: 'Swift',
            section: 'LngLat',
            location: { pathname: '/dr-ui/feedback/', hash: '#lnglat' },
            type: 'page',
            feedbackSentryDsn: 'https://abce@sentry.io/123'
          }
        })
      ).toMatchSnapshot();
    });

    test('3 - Something is missing', () => {
      expect(
        createSegmentEvent({
          state: {
            anonymousId: '7a6b4852-bacf-40e1-ae42-d91c4003f464',
            sessionId: '1624292376749-7a6b4852-bacf-40e1-ae42-d91c4003f464',
            isOpen: true,
            category: 'Report a problem',
            categoryType: 'Something is missing',
            feedback: 'My mind!',
            sentFeedback: true,
            contactSupport: false,
            helpful: false
          },
          props: {
            site: 'dr-ui',
            webhook: {
              staging: 'webhook',
              production: 'webhook'
            },
            preferredLanguage: 'Swift',
            section: 'LngLat',
            location: { pathname: '/dr-ui/feedback/', hash: '#lnglat' },
            type: 'page',
            feedbackSentryDsn: 'https://abce@sentry.io/123'
          }
        })
      ).toMatchSnapshot();
    });
  });

  describe('logged in user', () => {
    test('1 - Open feedback', () => {
      expect(
        createSegmentEvent({
          state: {
            anonymousId: '7a6b4852-bacf-40e1-ae42-d91c4003f464',
            user: {
              id: 'friend-of-the-docs',
              plan: { id: 'starter' }
            },
            sessionId: '1624292376749-7a6b4852-bacf-40e1-ae42-d91c4003f464',
            isOpen: true,
            sentFeedback: false,
            contactSupport: false
          },
          props: {
            site: 'dr-ui',
            webhook: { staging: 'webhook', production: 'webhook' },
            preferredLanguage: 'Swift',
            section: 'LngLat',
            location: { pathname: '/dr-ui/feedback/', hash: '#lnglat' },
            type: 'page',
            feedbackSentryDsn: 'https://abce@sentry.io/123'
          }
        })
      ).toMatchSnapshot();
    });

    test('2 - Report a problem', () => {
      expect(
        createSegmentEvent({
          state: {
            anonymousId: '7a6b4852-bacf-40e1-ae42-d91c4003f464',
            sessionId: '1624292376749-7a6b4852-bacf-40e1-ae42-d91c4003f464',
            user: {
              id: 'friend-of-the-docs',
              plan: { id: 'starter' }
            },
            isOpen: true,
            category: 'Report a problem',
            sentFeedback: false,
            contactSupport: false,
            helpful: false
          },
          props: {
            site: 'dr-ui',
            webhook: {
              staging: 'webhook',
              production: 'webhook'
            },
            preferredLanguage: 'Swift',
            section: 'LngLat',
            location: { pathname: '/dr-ui/feedback/', hash: '#lnglat' },
            type: 'page',
            feedbackSentryDsn: 'https://abce@sentry.io/123'
          }
        })
      ).toMatchSnapshot();
    });

    test('3 - Something is missing', () => {
      expect(
        createSegmentEvent({
          state: {
            anonymousId: '7a6b4852-bacf-40e1-ae42-d91c4003f464',
            sessionId: '1624292376749-7a6b4852-bacf-40e1-ae42-d91c4003f464',
            user: {
              id: 'friend-of-the-docs',
              plan: { id: 'starter' }
            },
            isOpen: true,
            category: 'Report a problem',
            categoryType: 'Something is missing',
            feedback: 'My mind!',
            sentFeedback: true,
            contactSupport: false,
            helpful: false
          },
          props: {
            site: 'dr-ui',
            webhook: {
              staging: 'webhook',
              production: 'webhook'
            },
            preferredLanguage: 'Swift',
            section: 'LngLat',
            location: { pathname: '/dr-ui/feedback/', hash: '#lnglat' },
            type: 'page',
            feedbackSentryDsn: 'https://abce@sentry.io/123'
          }
        })
      ).toMatchSnapshot();
    });
  });
});
