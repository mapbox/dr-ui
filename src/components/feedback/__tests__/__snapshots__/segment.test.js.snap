// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createSegmentEvent anonymous user 1 - Open feedback 1`] = `
Object {
  "anonymousId": "7a6b4852-bacf-40e1-ae42-d91c4003f464",
  "event": "Sent docs feedback",
  "properties": Object {
    "anonymousId": "7a6b4852-bacf-40e1-ae42-d91c4003f464",
    "category": undefined,
    "categoryType": undefined,
    "environment": "staging",
    "helpful": undefined,
    "page": Object {
      "hash": "#lnglat",
      "pathname": "/dr-ui/feedback/",
    },
    "section": "LngLat",
    "sessionId": "1624292376749-7a6b4852-bacf-40e1-ae42-d91c4003f464",
    "site": "dr-ui",
  },
}
`;

exports[`createSegmentEvent anonymous user 2 - Report a problem 1`] = `
Object {
  "anonymousId": "7a6b4852-bacf-40e1-ae42-d91c4003f464",
  "event": "Sent docs feedback",
  "properties": Object {
    "anonymousId": "7a6b4852-bacf-40e1-ae42-d91c4003f464",
    "category": "Report a problem",
    "categoryType": undefined,
    "environment": "staging",
    "helpful": false,
    "page": Object {
      "hash": "#lnglat",
      "pathname": "/dr-ui/feedback/",
    },
    "section": "LngLat",
    "sessionId": "1624292376749-7a6b4852-bacf-40e1-ae42-d91c4003f464",
    "site": "dr-ui",
  },
}
`;

exports[`createSegmentEvent anonymous user 3 - Something is missing 1`] = `
Object {
  "anonymousId": "7a6b4852-bacf-40e1-ae42-d91c4003f464",
  "event": "Sent docs feedback",
  "properties": Object {
    "anonymousId": "7a6b4852-bacf-40e1-ae42-d91c4003f464",
    "category": "Report a problem",
    "categoryType": "Something is missing",
    "environment": "staging",
    "feedback": "My mind!",
    "helpful": false,
    "page": Object {
      "hash": "#lnglat",
      "pathname": "/dr-ui/feedback/",
    },
    "section": "LngLat",
    "sessionId": "1624292376749-7a6b4852-bacf-40e1-ae42-d91c4003f464",
    "site": "dr-ui",
  },
}
`;

exports[`createSegmentEvent logged in user 1 - Open feedback 1`] = `
Object {
  "event": "Sent docs feedback",
  "properties": Object {
    "category": undefined,
    "categoryType": undefined,
    "environment": "staging",
    "helpful": undefined,
    "page": Object {
      "hash": "#lnglat",
      "pathname": "/dr-ui/feedback/",
    },
    "plan": "starter",
    "section": "LngLat",
    "sessionId": "1624292376749-7a6b4852-bacf-40e1-ae42-d91c4003f464",
    "site": "dr-ui",
    "userId": "friend-of-the-docs",
  },
  "userId": "friend-of-the-docs",
}
`;

exports[`createSegmentEvent logged in user 2 - Report a problem 1`] = `
Object {
  "event": "Sent docs feedback",
  "properties": Object {
    "category": "Report a problem",
    "categoryType": undefined,
    "environment": "staging",
    "helpful": false,
    "page": Object {
      "hash": "#lnglat",
      "pathname": "/dr-ui/feedback/",
    },
    "plan": "starter",
    "section": "LngLat",
    "sessionId": "1624292376749-7a6b4852-bacf-40e1-ae42-d91c4003f464",
    "site": "dr-ui",
    "userId": "friend-of-the-docs",
  },
  "userId": "friend-of-the-docs",
}
`;

exports[`createSegmentEvent logged in user 3 - Something is missing 1`] = `
Object {
  "event": "Sent docs feedback",
  "properties": Object {
    "category": "Report a problem",
    "categoryType": "Something is missing",
    "environment": "staging",
    "feedback": "My mind!",
    "helpful": false,
    "page": Object {
      "hash": "#lnglat",
      "pathname": "/dr-ui/feedback/",
    },
    "plan": "starter",
    "section": "LngLat",
    "sessionId": "1624292376749-7a6b4852-bacf-40e1-ae42-d91c4003f464",
    "site": "dr-ui",
    "userId": "friend-of-the-docs",
  },
  "userId": "friend-of-the-docs",
}
`;
